File: device/devtitans/kraken_ir/KrakenIRapp/app/src/main/java/com/devtitans/KrakenIRapp/database/IRCommandDatabase.java
package com.devtitans.KrakenIRapp.database;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.devtitans.KrakenIRapp.models.IRCommand;

@Database(entities = {IRCommand.class}, version = 1, exportSchema = false)
public abstract class IRCommandDatabase extends RoomDatabase {

    public abstract IRCommandDAO getNoteDao();

    private static volatile IRCommandDatabase INSTANCE;

    public static IRCommandDatabase getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (IRCommandDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                                    IRCommandDatabase.class, "ir_command_database")
                            .build();
                }
            }
        }
        return INSTANCE;
    }
}

File: device/devtitans/kraken_ir/IRApp/app/src/main/java/com/wax/irapp/database/IRCommandDatabase.kt
package com.wax.irapp.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.wax.irapp.models.IRCommand

@Database(entities = arrayOf(IRCommand::class), version = 1, exportSchema = false)
abstract class IRCommandDatabase : RoomDatabase() {

    abstract fun getNoteDao() : IRCommandDAO

    companion object{

        @Volatile
        private var INSTANCE : IRCommandDatabase? = null

        fun getDatabase(context : Context) : IRCommandDatabase{
            return INSTANCE ?: synchronized(this){

                val instance = Room.databaseBuilder(
                    context.applicationContext,
                    IRCommandDatabase::class.java,
                    "ir_command_database"
                ).build()

                INSTANCE = instance

                instance
            }
        }
    }

}